annotation/tests/test_annotations.py

```diff
@@ -482,3 +588,491 @@ def test_construct_annotated_doc_db_error(
+        mock_load_page.assert_has_calls(
+            [
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    1,
+                    expected_user_1,
+                    expected_page_rev_1,
+                    True,
+                ),
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    2,
+                    expected_user_2,
+                    expected_page_rev_2,
+                    True,
+                ),
+            ]
+        )
+        assert mock_load_page.call_count == 2
+    assert page_revision_list_all == expected_pages
+
+
+def test_load_latest_revs_pages(
+    page_revision_list_latest: Dict[int, Dict[str, LatestPageRevision]]
+):
+    expected_user_1 = list(page_revision_list_latest[1].keys())[0]
+    expected_user_2 = list(page_revision_list_latest[2].keys())[0]
+    expected_page_rev_1 = {**page_revision_list_latest[1][expected_user_1]}
+    expected_page_rev_2 = {**page_revision_list_latest[2][expected_user_2]}
+    expected_pages = {1: [], 2: []}
+
+    with patch(
+        "annotation.annotations.main.convert_bucket_name_if_s3prefix",
+        return_value=TEST_TENANT,
+    ) as mock_convert_prefix, patch(
+        "annotation.annotations.main.load_page"
+    ) as mock_load_page:
+        load_latest_revision_pages(page_revision_list_latest, TEST_TENANT)
+        mock_convert_prefix.assert_called_once_with(TEST_TENANT)
+        mock_load_page.assert_has_calls(
+            [
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    1,
+                    expected_user_1,
+                    expected_page_rev_1,
+                    True,
+                ),
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    2,
+                    expected_user_2,
+                    expected_page_rev_2,
+                    True,
+                ),
+            ]
+        )
+        assert mock_load_page.call_count == 2
+    assert page_revision_list_latest == expected_pages
+
+
+def test_load_annotated_pages_for_particular_rev(
+    annotated_doc: AnnotatedDoc,
+    page_revision_list_all: Dict[int, List[PageRevision]],
+):
+    annotated_doc.pages[1] = "sha1"
+    with patch("annotation.annotations.main.load_page") as mock_load_page:
+        load_annotated_pages_for_particular_rev(
+            annotated_doc, page_revision_list_all[1][0], TEST_TENANT, []
+        )
+        mock_load_page.assert_called_once_with(
+            None,
+            [],
+            TEST_TENANT,
+            1,
+            annotated_doc.user,
+            page_revision_list_all[1][0],
+            False,
+        )
+    assert page_revision_list_all[1][0]["page_id"] == "sha1"
+
+
+def test_load_validated_pages_for_particular_rev(
```

Comment:
Function utilize a logger, you can mock the logger to verify that the correct messages are being logged.

Reply:
Thanks, done